alarm_control_panel/elkm1.py

```diff
@@ -207,59 +208,32 @@ def trigger_update(self, attribute, value):
     @asyncio.coroutine
     def async_update(self):
         """Get the latest data and update the state."""
-        from elkm1.const import ArmedStatus, AlarmState
-        # Set status based on arm state
-        self._armed_status = self._element.armed_status
-        if self._armed_status is not None:
-            if self._armed_status == ArmedStatus.DISARMED.value:
-                self._state = STATE_ALARM_DISARMED
-            elif self._armed_status == ArmedStatus.ARMED_AWAY.value:
-                self._state = STATE_ALARM_ARMED_AWAY
-            elif self._armed_status == ArmedStatus.ARMED_STAY.value:
-                self._state = STATE_ALARM_ARMED_HOME
-            elif self._armed_status == ArmedStatus.ARMED_STAY_INSTANT.value:
-                self._state = STATE_ALARM_ARMED_HOME
-            elif self._armed_status == ArmedStatus.ARMED_TO_NIGHT.value:
-                self._state = STATE_ALARM_ARMED_HOME
-            elif self._armed_status == ArmedStatus.ARMED_TO_NIGHT_INSTANT.value:
-                self._state = STATE_ALARM_ARMED_HOME
-            elif self._armed_status == ArmedStatus.ARMED_TO_VACATION.value:
-                self._state = STATE_ALARM_ARMED_AWAY
-        else:
+
+        if self._element.alarm_state is None:
             self._state = STATE_UNKNOWN
-        # If alarm is triggered, show that instead
-        if self._element.alarm_state is not None:
-            if self._element.alarm_state != AlarmState.NO_ALARM_ACTIVE.value:
-                self._state = STATE_ALARM_TRIGGERED
-        # Unless there's an entry / exit timer running,
-        # show that we're arming or pending alarm accordingly
-        if self._element.timer1 > 0 or self._element.timer2 > 0:
-            if not self._element.is_exit:
-                self._state = STATE_ALARM_PENDING
-            # Don't displaying ARMING if exit timer running, because
-            # HASS won't let you disarm during ARMING
-            #else:
-            #    self._state = STATE_ALARM_ARMING
-        # If we should be hidden due to lack of member devices and default name, hide us
-        if (len(self._keypads) == 0) and (len(self._zones) == 0) and (self._element.is_default_name()):
-            self._hidden = True
+        elif self._element.timer1 > 0 or self._element.timer2 > 0:
+            self._state = STATE_ALARM_ARMING if self._element.is_exit \
+                else STATE_ALARM_DISARMING
+            # Temporary fix until old UI arm dialog fixed
+            self._state = STATE_ALARM_PENDING
+        elif self._element.alarm_state >= AlarmState.FIRE_ALARM.value:
```

Comment:
I'll have to test and check to verify, but might have to reverse the order of `elif self._element.timer ...` and `elif self._element.alarm_state >= ...` (as I had them before) to ensure that if timers are running, we aren't triggering the alarmed state in HASS immediately (which would overwrite the arming/disarming/pending state)